pkg/registry/networking/networkpolicy/strategy.go

```diff
@@ -107,6 +119,62 @@ func (networkPolicyStrategy) AllowUnconditionalUpdate() bool {
 	return true
 }
 
+type networkPolicyStatusStrategy struct {
+	networkPolicyStrategy
+}
+
+// StatusStrategy implements logic used to validate and prepare for updates of the status subresource
+var StatusStrategy = networkPolicyStatusStrategy{Strategy}
+
+// GetResetFields returns the set of fields that get reset by the strategy
+// and should not be modified by the user.
+// TODO: Am I doing right here? Netpol didn't had a ResetFields on the specStrategu so I'm assuming this only applies for status here?
+func (networkPolicyStatusStrategy) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
+	fields := map[fieldpath.APIVersion]*fieldpath.Set{
+		"extensions/v1beta1": fieldpath.NewSet(
+			fieldpath.MakePathOrDie("status", "conditions"),
+		),
+		"networking.k8s.io/v1": fieldpath.NewSet(
+			fieldpath.MakePathOrDie("status", "conditions"),
+		),
+	}
+
+	return fields
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update of status
+func (networkPolicyStatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newNetworkPolicy := obj.(*networking.NetworkPolicy)
+	oldNetworkPolicy := old.(*networking.NetworkPolicy)
+	// status changes are not allowed to update spec
+	newNetworkPolicy.Spec = oldNetworkPolicy.Spec
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.NetworkPolicyStatus) {
+
+		// As network policy status is composed only of an array of conditions, we can say that the status
+		// is in use if the condition array is bigger than 0.
+		if len(oldNetworkPolicy.Status.Conditions) < 1 {
+			newNetworkPolicy.Status = networking.NetworkPolicyStatus{}
+		}
+		// TODO: What to do when status field is in usage and we revert FG? Drop? Keep the old value? Allow updates?
```

Comment:
Yes, we generally keep data in this case, but no updates except to clear it.  It's imperfect but rare and complicated to get right (arguably impossible since the implications of resetting the data could be arbitrary)